app/views/slices/firstParagraph.php

```diff
@@ -1,29 +1,29 @@
 <?php
+use Prismic\Dom\RichText;
 
 // Get all the slices of the post
-$sliceZone = $post->getSliceZone('post.body');
+$slices = $post->data->body;
 $firstParagraph = "";
 
 // Loop through each slice to find the first text slice
-foreach ($sliceZone->getSlices() as $slice) {
-  if ($slice->getSliceType() == "text") {
-    $firstParagraph = $slice->getPrimary()->getStructuredText('text')->getFirstParagraph();
+foreach ($slices as $slice) {
+  if ($slice->slice_type == "text") {
+    $firstText = RichText::asHtml($slice->primary->text, $linkResolver);
```

Comment:
I think you should just get the Rich Text `asText`. In the case where it goes over the limit of 300 characters, this will cut off the closing HTML elements.

Reply:
As discussed earlier, while asHtml adds some nice visualization of html formatting in the blog homepage, it introduces the problem of broken tags if the closing is outside the 300 character limit. Falling back to asText for clarity and avoiding first user problems.